ansible_collections/arista/avd/plugins/action/anta_runner.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+import json
+
+from ansible.plugins.action import ActionBase
+from anta.inventory.models import InventoryDevice
+from anta.result_manager import ResultManager
+
+
+class ActionModule(ActionBase):
+    def run(self, tmp=None, task_vars=None):
+        if task_vars is None:
+            task_vars = {}
+
+        result = super().run(tmp, task_vars)
+        del tmp  # tmp no longer has any effect
+
+        device_params = {
+            "name": task_vars["inventory_hostname"],
+            "host": task_vars["ansible_host"],
+            "username": task_vars["ansible_user"],
+            "password": task_vars["ansible_httpapi_password"],
+            "port": 443,
+            "session": self._connection.httpapi,
+        }
+
+        try:
+            anta_device = InventoryDevice(**device_params)
+            tests_catalog = task_vars["anta_catalog"]["anta_catalog"]
+            results = ResultManager()
+
+            for test_class, test_params in tests_catalog:
+                test_instance = test_class(device=anta_device, from_ansible=True)
+                for index, command in enumerate(test_instance.instance_commands):
+                    response = anta_device.session.send_request(command.command, version=command.version, output=command.ofmt)
```

Comment:
So with this you basically moved the device interaction out of ANTA and loop over them here. It is fine for now, but I think we want it to be done/maintained inside ANTA instead, but using a passed interface to httpapi.